src/generators/result.gen.ts

```diff
@@ -0,0 +1,237 @@
+
+function getAllMethodsAndAddOrigin(file: ts.SourceFile, name: string, isInterface = false){
+  return getAllMethodsByFileAndName(file, checker, name, isInterface)
+    .map((node) =>
+      ({
+        ...node,
+        origin: (interfaceName !== name) ? name : generatedClassName
+      }));
+}
+
+const asyncMethods = getAllMethodsAndAddOrigin(asyncSourceFile!, asyncClassName).filter((e) => !e.hasImplementation);
+const syncMethods = getAllMethodsAndAddOrigin(syncSourceFile!, syncClassName).filter((e) => !e.hasImplementation);
+const resultMethods = getAllMethodsAndAddOrigin(sourceFile!, interfaceName, true);
+
+const abstractMethods = resultMethods.map(e => `${e.documentation.raw?.fullRaw ?? ''}\nabstract ${e.content}`).join('\n')
+
+const signatures = [
+  ...asyncMethods,
+  ...syncMethods,
+  ...resultMethods
+]
+
+const signaturesGrouped = mergeFunctionOverloadDocumentation(_.groupBy(signatures, 'name'));
+
+const funcs = _.map(signaturesGrouped, (value, name) => {
+  let paramsCount = 0;
+  let types: Record<string, any> = {}
+  let comment: string | undefined;
+  _.forEach(value, (signature) => {
+    const { generics, parameters, returnType, origin } = signature;
+    paramsCount = Math.max(paramsCount, parameters.length);
+    comment = signature.documentation.method;
+
+    signature.documentation.fields = {
+      ...signature.documentation.fields,
+      ...{ [instanceParam[0]]: instanceParameterComment },
+    };
+
+    const newOverload = createTypeOverload(
+      [
+        ...parameters,
+        {
+          name: instanceParam[0],
+          type: `${origin}<T>`,
+          consumedTypes: instanceParam[2],
+          documentation: instanceParameterComment,
+          isOptional: false,
+        },
+      ],
+      [
+        //X extends AsyncResult<any> = AsyncResult<any>, T = X extends Result<infer U> ? U : any
+        { name: 'T', fullValue: 'T = any' },
+        ...generics,
+        // { name: 'X', fullValue: `X extends ${origin}<any> = ${origin}<any>` },
+      ],
+      returnType.type,
+      signature.documentation
+    );
+
+    types = merge(types, newOverload);
+
+  });
+
+
+  return templatefuncImplementation(
+    name,
+    TypeOverloadByParametersObjToStr(types),
+    paramsCount + 1,
+    comment
+  );
+}).join('');
+
+const fileContents = `
+    ${warning}
+    ${importStatments}
+    ${classComment}
+    ${classTemplate(generatedClassName, `${baseFunctions}\n${funcs}\n${abstractMethods}`)}
+`;
+
+const resultFile = ts.createSourceFile(
+  generatedFileName,
+  fileContents,
+  ts.ScriptTarget.ESNext,
+  false,
+  ts.ScriptKind.TS
+);
+const printer = ts.createPrinter({ newLine: ts.NewLineKind.LineFeed });
+
+ts.sys.writeFile(
+  `${generatedDirectoryPath}/${resultFile.fileName}`,
+  printer.printFile(resultFile)
+);
+
+
+function mergeFunctionOverloadDocumentation(groupedFunctions: _.Dictionary<(MethodDefinition & { origin: string })[]>){
+
+  _.forEach(groupedFunctions, (value, name) => {

```

Comment:
Another newline.

In order to reduce the dependencies on lodash and thus make the package smaller, is it possible to use JS' own `forEach` instead of lodash's?

Reply:
Lodash is only required as a dev-dependency, the only dependency for lonad 'proper' is `lodash.flow` which doesn't require lodash.